src/pg/filter/getSql.js

```diff
@@ -45,6 +47,8 @@ function getBinarySql(lhs, type, op, value, textLhs) {
     if (typeof value === 'string') {
       return sql`${textLhs} ${sqlOp} ${value}`;
     }
+    if ((op === '$keycts' || op === '$keyctd') && Array.isArray(value))
+        return sql`${lhs} ${sqlOp} array[${join(value)}]`;
```

Comment:
Could you check with just:
```suggestion
        return sql`${lhs} ${sqlOp} ${value}::text[]`;
```

If that works, it will only use one placeholder for the whole array rather than separate ones for each item, which will cut query planning time a lot.

Reply:
Please try out this actual code change. IIUC, when the value is an array (say `['foo', 'bar']`) node-postgres will generate send the SQL value `{"foo","bar"}` which is one PostgreSQL literal syntax for a text array.